invenio_records_rest/serializers/csv.py

```diff
@@ -0,0 +1,109 @@
+        """Initialize CSVSerializer.
+
+        :param csv_excluded_fields: list of complete paths of the fields that
+                                    should be excluded from the final output
+        """
+        self.csv_excluded_fields = kwargs.pop("csv_excluded_fields", [])
+        super(CSVSerializer, self).__init__(*args, **kwargs)
+
+    def serialize(self, pid, record, links_factory=None):
+        """Serialize a single record and persistent identifier.
+
+        :param pid: Persistent identifier instance.
+        :param record: Record instance.
+        :param links_factory: Factory function for record links.
+        """
+        record = self.process_dict(
+            self.transform_record(pid, record, links_factory))
+
+        return self.format_csv([record])
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        :param item_links_factory: Factory function for record links.
+        """
+        records = []
+        for hit in search_result['hits']['hits']:
+            processed_hit = self.transform_search_hit(
+                pid_fetcher(hit['_id'], hit['_source']),
+                hit,
+                links_factory=item_links_factory,
+            )
+            records.append(self.process_dict(processed_hit))
+
+        return self.format_csv(records)
+
+    def format_csv(self, records):
+        """Format list of flattened dictionaries into CSV format."""
+        # build a list of all the headers from all the rows
+        # and remove any duplicates with a set
+        headers = sorted(set([key for record in records for key in record]))
+
+        csv_string = ",".join(map(str, headers)) + "\n"
+        newline = ""
+        for record in records:
+            separator = ""
+            csv_string += newline
+            for key in headers:
+                csv_string += separator
+                csv_string += "\"{}\"".format(
+                    record[key]) if key in record else ""
+                separator = ","
+            newline = "\n"
+        return csv_string
+
+    def process_dict(self, dictionary):
+        """Returns a flattened dictionary as a string."""
+        flattened = {}
+
+        self.flatten(dictionary, flattened)
+
+        for path in self.csv_excluded_fields:
+            for key in list(flattened.keys())[:]:
+                if path in key:
+                    del flattened[key]
+
+        return flattened
+
+    def flatten(self, elem, flattened, parent_key=''):
+        """Flattens nested dictionaries."""
+        if type(elem) is dict:
+            for key in elem:
+                self.flatten(elem[key], flattened, parent_key + key + '_')
+        elif type(elem) is list:
```

Comment:
I've always found more pythonic something like `isinstance(elem, MutableSequence)` but I guess it's just personal preference ☺️